src/main/java/uk/gov/ons/census/casesvc/service/AddressModificationService.java

```diff
@@ -0,0 +1,107 @@
+package uk.gov.ons.census.casesvc.service;
+
+import static uk.gov.ons.census.casesvc.utility.JsonHelper.convertObjectToJson;
+
+import java.time.OffsetDateTime;
+import java.util.Set;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+import uk.gov.ons.census.casesvc.logging.EventLogger;
+import uk.gov.ons.census.casesvc.model.dto.AddressModification;
+import uk.gov.ons.census.casesvc.model.dto.ResponseManagementEvent;
+import uk.gov.ons.census.casesvc.model.entity.Case;
+import uk.gov.ons.census.casesvc.model.entity.EventType;
+
+@Service
+public class AddressModificationService {
+  private static final Set<String> ESTAB_TYPES =
+      Set.of(
+          "HALL OF RESIDENCE",
+          "CARE HOME",
+          "HOSPITAL",
+          "HOSPICE",
+          "MENTAL HEALTH HOSPITAL",
+          "MEDICAL CARE OTHER",
+          "BOARDING SCHOOL",
+          "LOW/MEDIUM SECURE MENTAL HEALTH",
+          "HIGH SECURE MENTAL HEALTH",
+          "HOTEL",
+          "YOUTH HOSTEL",
+          "HOSTEL",
+          "MILITARY SLA",
+          "MILITARY US",
+          "RELIGIOUS COMMUNITY",
+          "RESIDENTIAL CHILDRENS HOME",
+          "EDUCATION OTHER",
+          "PRISON",
+          "IMMIGRATION REMOVAL CENTRE",
+          "APPROVED PREMISES",
+          "ROUGH SLEEPER",
+          "STAFF ACCOMMODATION",
+          "CAMPHILL",
+          "HOLIDAY PARK",
+          "HOUSEHOLD",
+          "SHELTERED ACCOMMODATION",
+          "RESIDENTIAL CARAVAN",
+          "RESIDENTIAL BOAT",
+          "GATED APARTMENTS",
+          "MOD HOUSEHOLDS",
+          "FOREIGN OFFICES",
+          "CASTLES",
+          "GRT SITE",
+          "MILITARY SFA",
+          "EMBASSY",
+          "ROYAL HOUSEHOLD",
+          "CARAVAN SITE",
+          "MARINA",
+          "TRAVELLING PERSONS",
+          "TRANSIENT PERSONS");
+
+  private final CaseService caseService;
+  private final EventLogger eventLogger;
+
+  public AddressModificationService(CaseService caseService, EventLogger eventLogger) {
+    this.caseService = caseService;
+    this.eventLogger = eventLogger;
+  }
+
+  public void processMessage(
+      ResponseManagementEvent responseManagementEvent, OffsetDateTime messageTimestamp) {
+    AddressModification addressModification =
+        responseManagementEvent.getPayload().getAddressModification();
+
+    validate(addressModification);
+
+    Case caze = caseService.getCaseByCaseId(addressModification.getCollectionCase().getId());
+
+    caze.setEstabType(addressModification.getNewAddress().getEstabType());
+    caze.setTownName(addressModification.getNewAddress().getTownName());
+    caze.setAddressLine1(addressModification.getNewAddress().getAddressLine1());
+    caze.setAddressLine2(addressModification.getNewAddress().getAddressLine2());
+    caze.setAddressLine3(addressModification.getNewAddress().getAddressLine3());
+    caze.setOrganisationName(addressModification.getNewAddress().getOrganisationName());
```

Comment:
Would this cope with being able to modify only a subset of the fields? It looks like if they send us a message with, for example, a missing or `null` organisation name this would null it on the case, I don't see a way to leave these fields out of modification

Reply:
we might be able to use `Optional` in the DTO... gonna do an experiment

Reply:
have a look now. I don't think it implements the requirement (process address modification messages) any better than it did before, because CC/FWMT-G are not going to send us hand-crafted JSON with only the unique bits that have changed - they're just going to splat the whole address because that's easiest

Reply:
If we can guaranty we're always going to be sent everything then I'd agree it's unnecessary, but since the purpose of this is vanity changes I was under the impression they would be trying to update minimal fields. I also thought we were avoiding asking people to supply the current, unchanged data because that opens the possibility of accidentally reverting other changes. 

Do we need to clarify exactly how this is intended to be used before proceeding with this? The example in the event dictionary still has an `oldAddress` block, but does also leave `townName` out of the `newAddress` so I'm not sure what to make of it.